tests/integration/api_v3/http/generators.test.js

```diff
@@ -36,16 +36,23 @@ const STATUS = {
 describe('Generators API', () => {
 	let numberActiveValidators;
 	let numberStandbyValidators;
-	beforeAll(async () => {
-		const response = (await api.get(`${endpoint}/pos/constants`)).data;
-		numberActiveValidators = response.numberActiveValidators;
-		numberStandbyValidators = response.numberStandbyValidators;
-	});
+	let selectedGenerator;
 
-	let firstGenerator;
 	beforeAll(async () => {
+		const posRes = (await api.get(`${endpoint}/pos/constants`)).data;
+		numberActiveValidators = posRes.numberActiveValidators;
+		numberStandbyValidators = posRes.numberStandbyValidators;
+
 		const response = await api.get(`${endpoint}/generators`);
-		[firstGenerator] = response.data;
+		[selectedGenerator] = response.data;
+
+		// eslint-disable-next-line no-restricted-syntax
+		for (const generator of response.data) {
+			if (generator.publicKey !== null && generator.publicKey !== undefined) {
```

Comment:
May be we can update it to
```suggestion
			if (![null, undefined].includes(generator.publicKey)) {
```